test-framework/common/src/main/java/io/quarkus/test/common/WithTestResource.java

```diff
@@ -13,11 +13,20 @@
 /**
  * Used to define a test resource, which can affect various aspects of the application lifecycle.
  * <p>
+ * As of Quarkus 3.16, the default behavior of the annotation (meaning that {@code scope} has not been set)
+ * is that test classes annotated with same {@code WithTestResource} will <strong>not</strong> force a restart
```

Comment:
```suggestion
 * is that test classes annotated with the same {@code WithTestResource} will <strong>not</strong> force a restart
```

I feel like "the same test resource" should mention that they also must have the same config, no?